### PR TITLE
remove control plane items off version cmd

### PR DESCRIPTION
**Pull request by linsun**
attempt to fix https://github.com/istio/istio/issues/18937 to avoid confusion

### CODE REVIEW

File: istioctl/cmd/version.go [hunk 1]
```diff
@@ -98,10 +99,14 @@ func getProxyInfo() (*[]istioVersion.ProxyInfo, error) {
 		}
 
 		for _, ss := range sss {
-			pi = append(pi, istioVersion.ProxyInfo{
-				ID:           ss.ProxyID,
-				IstioVersion: ss.SyncStatus.IstioVersion,
-			})
+			// if proxy id has istio gateway items, let us exclude them
+			// from data plane as they are also shown under control plane
+			if !strings.HasPrefix(ss.ProxyID, "istio-ingressgateway-") && !strings.HasPrefix(ss.ProxyID, "istio-egressgateway-") && !strings.HasPrefix(ss.ProxyID, "prometheus-") {
```
Comment (howardjohn) [inline istioctl/cmd/version.go]: istio-ingressgateway is not a special name, we just happen to name it this way. These are very much a part of the data plane. If anything, we should remove them from the control plane list in my opinion.

Comment (linsun) [reply istioctl/cmd/version.go]: I do agree the gateways are part of the data plane conceptually, however from the upgrade perspective, users are thinking they are part of Istio and updated as part of Istio control plane update per our doc:  https://istio.io/docs/setup/upgrade/istioctl-upgrade/#upgrade-steps. cc @kramvan1 @esnible for input

Comment (linsun) [reply istioctl/cmd/version.go]: To me, the istioctl version cmd for the data plane section is mostly used for evaluating if the data plane sidecars need to be updated.

**Comment by linsun** [comment]
@esnible @howardjohn PTAL - I've updated to remove istio gateways off control plane list when `istioctl version --short=false` is used.

**Comment by esnible** [comment]
I am testing this now.

My first thought was that the PR should remove the Ingress/Egress/ILB from the sidecar list.  You are removing it from the control plane list.

It is a nice simple implementation.

Before I give this a 👍 do you think the Ingress/Egress/ILB should be with the data plane counts?  By putting them with the control plane we separate the Istio installation from sidecars running on user workloads.

**Comment by linsun** [comment]
Thank you @esnible, BTW, here is my test result:

```
$ out/darwin_amd64/istioctl version --short=false
client version: version.BuildInfo{Version:"5ea2def4d453e23fc74f93447ecbc9b5d9272c83-dirty", GitRevision:"5ea2def4d453e23fc74f93447ecbc9b5d9272c83-dirty", GolangVersion:"go1.14.2", BuildStatus:"Modified", GitTag:"1.6.0-alpha.1-125-g5ea2def4d"}
pilot version: version.BuildInfo{Version:"1.5.1", GitRevision:"9d07e185b0dd50e6fb1418caa4b4d879788807e3", GolangVersion:"go1.13.8", BuildStatus:"Clean", GitTag:"1.5.1"}
data plane version: version.ProxyInfo{ID:"istio-ingressgateway-5d6c9f74cb-d9rv9.istio-system", IstioVersion:"1.5.1"}
data plane version: version.ProxyInfo{ID:"helloworld-v2-776f74c475-crqws.sample", IstioVersion:"1.5.1"}
data plane version: version.ProxyInfo{ID:"prometheus-58484748c4-5h4c2.istio-system", IstioVersion:"1.5.1"}
data plane version: version.ProxyInfo{ID:"sleep-f8cbf5b76-psl4f.sample", IstioVersion:"1.5.1"}
data plane version: version.ProxyInfo{ID:"traderv2-78d979d68-9prxw.stock-trader", IstioVersion:"1.5.1"}
data plane version: version.ProxyInfo{ID:"trader-56f9fcbbb7-g4scd.stock-trader", IstioVersion:"1.5.1"}
data plane version: version.ProxyInfo{ID:"stock-quote-5f7bf486db-wg6jf.stock-trader", IstioVersion:"1.5.1"}
data plane version: version.ProxyInfo{ID:"portfolio-7bd45cf5f8-ppvkq.stock-trader", IstioVersion:"1.5.1"}
```

My initial PR was to remove the gateways off data plane but I got rejection from @howardjohn and @liamawhite.  Think it further, i think it makes sense to include it from data plane because it is part of the data plane.  It also helps with the scenario like https://github.com/istio/istio/issues/22934 where you won't have istiod on remote cluster but you will still have gateways there.

**Pull request merged**